### PR TITLE
Bug fix for searching through multiple repositories, some of which do not contain pkg to find

### PR DESCRIPTION
**Pull request by alerickson**
# PR Summary
When `Find-PSResource` searches through multiple repositories, it was previously writing errors for each repository it could not find the given package in along side the package information from the repository where it could find it.

If a package does exist in at least one repository, the changes in this PR will not write errors from the repositories where the package does not exist and instead writes verbose messages and still display package info for the repo where it is found.

If a package does not exist in any repositories, a message that states: "Package(s) '<PackageName>' could not be found in any registered repositories."  will be displayed, along with any other errors unrelated to the package not being found (such as unauthorized exceptions).

## PR Context

Resolves the first part of issue #1286 

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [x] None
    - **OR**
    - [ ] [Documentation needed]()
        - [ ] Issue filed: 
- **User-facing changes**
    - [ ] Not Applicable
    - **OR**
    - [x] [Documentation needed]()
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [ ] N/A or can only be tested interactively
    - **OR**
    - [x] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [x] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and enumerated concerns in the summary.

### CODE REVIEW

File: src/code/FindHelper.cs [hunk 1]
```diff
@@ -152,8 +153,19 @@ public FindHelper(CancellationToken cancellationToken, PSCmdlet cmdletPassedIn,
 
                 foreach (PSResourceInfo currentPkg in SearchByNames(currentServer, currentResponseUtil, currentRepository))
                 {
+                    pkgFound = currentPkg != null;
```
Comment (adityapatwardhan) [inline src/code/FindHelper.cs]: It seems that currentPkg can be null here. Should we be yield returning a null then?
Should be continue instead when it is null?

File: src/code/FindHelper.cs [hunk 2]
```diff
@@ -152,8 +153,19 @@ public FindHelper(CancellationToken cancellationToken, PSCmdlet cmdletPassedIn,
...
                     yield return currentPkg;
-                }   
+                }
+            }
+
+            // Do not write out error message if -Name "*"
+            if (!pkgFound && !_pkgsLeftToFind.Contains("*"))
+            {
+                _cmdletPassedIn.WriteError(new ErrorRecord(
+                            new ResourceNotFoundException($"Package(s) '{string.Join(", ", _pkgsLeftToFind)}' cound not be found in any registered repositories."),
```
Comment (adityapatwardhan) [inline src/code/FindHelper.cs]:
```suggestion
                            new ResourceNotFoundException($"Package(s) '{string.Join(", ", _pkgsLeftToFind)}' could not be found in any registered repositories."),
```

File: src/code/FindHelper.cs [hunk 3]
```diff
@@ -424,19 +454,21 @@ private IEnumerable<PSResourceInfo> SearchByNames(ServerApiCall currentServer, R
...
                         foreach (PSResourceResult currentResult in currentResponseUtil.ConvertToPSResourceResult(responseResults: responses))
                         {
                             if (currentResult.exception != null && !currentResult.exception.Message.Equals(string.Empty))
                             {
-                                _cmdletPassedIn.WriteError(new ErrorRecord(
-                                            new PackageNotFoundException($"Package '{pkgName}' could not be found", currentResult.exception), 
-                                            "FindAllConvertToPSResourceFailure", 
-                                            ErrorCategory.InvalidResult, 
-                                            this));
+                               _cmdletPassedIn.WriteVerbose($"Package '{pkgName}' could not be found");
```
Comment (adityapatwardhan) [inline src/code/FindHelper.cs]: consider having repository name in the message here.

File: src/code/FindHelper.cs [hunk 4]
```diff
@@ -624,20 +675,28 @@ private IEnumerable<PSResourceInfo> SearchByNames(ServerApiCall currentServer, R
...
 
                         foreach (PSResourceResult currentResult in currentResponseUtil.ConvertToPSResourceResult(responses))
                         {
                             if (currentResult.exception != null && !currentResult.exception.Message.Equals(string.Empty))
                             {
-                                // write warning?
-                                _cmdletPassedIn.WriteError(new ErrorRecord(
-                                            new PackageNotFoundException($"Package '{pkgName}' with version '{_version}' could not be found", currentResult.exception), 
+                                errRecord = new ErrorRecord(
+                                            new ResourceNotFoundException($"Package '{pkgName}' with version '{_version}' could not be found", currentResult.exception),
```
Comment (adityapatwardhan) [inline src/code/FindHelper.cs]: same as above

**Review by adityapatwardhan — CHANGES_REQUESTED** [review]
Couple of suggestions above

File: src/code/FindHelper.cs (hunk 4 above)
Comment (alerickson) [reply src/code/FindHelper.cs]: Anam made these changes in a PR that got merged, so I'll just pull in those changes.

File: src/code/FindHelper.cs [hunk 5]
```diff
@@ -153,16 +155,23 @@ public FindHelper(CancellationToken cancellationToken, PSCmdlet cmdletPassedIn,
 
                 foreach (PSResourceInfo currentPkg in SearchByNames(currentServer, currentResponseUtil, currentRepository))
                 {
-                    pkgFound = currentPkg != null;
+                    if (currentPkg == null) {
+                        continue;
+                    }
+                    pkgFound = true;
+
```
Comment (adityapatwardhan) [inline src/code/FindHelper.cs]:
```suggestion

                    pkgFound = true;
```

File: src/code/FindHelper.cs [hunk 6]
```diff
@@ -469,7 +478,7 @@ private IEnumerable<PSResourceInfo> SearchByNames(ServerApiCall currentServer, R
                         {
                             if (currentResult.exception != null && !currentResult.exception.Message.Equals(string.Empty))
                             {
-                               _cmdletPassedIn.WriteVerbose($"Package '{pkgName}' could not be found");
+                                _cmdletPassedIn.WriteVerbose($"Package '{pkgName}' could not be found in repository '{repository.Name}");
```
Comment (adityapatwardhan) [inline src/code/FindHelper.cs]:
```suggestion
                               _cmdletPassedIn.WriteVerbose($"Package '{pkgName}' could not be found in repository '{repository.Name}");
```

**Review by adityapatwardhan — APPROVED** [review]

**Pull request merged**